test/test_autograd.py

```diff
@@ -5800,10 +5800,17 @@ def test(get_input, cuda, pin_memory):
                 self.assertEqual(a, y.grad_fn._saved_other)
                 self.assertEqual(a.dtype, y.grad_fn._saved_self.dtype)
                 self.assertEqual(a.layout, y.grad_fn._saved_self.layout)
-                if y.is_sparse:
+                if a.is_sparse:
```

Comment:
cc @IvanYashchuk @cpuhrsch @pearu 

Why the change from `y` to `a` in the conditional?

Reply:
Consistency. Below we can no longer check for `y` so I thought always checking for `a` might be more clear. No strong opinion, can revert.